pkg/plugins/manager/loader/finder/local.go

```diff
@@ -186,6 +186,17 @@ func (l *Local) getAbsPluginJSONPaths(path string, followDistFolder bool) ([]str
 				return util.ErrWalkSkipDir
 			}
 
+			//if fi.Name() == "dist" {
```

Comment:
I added this here in case having this logic as part of the WalkFunc is easier to follow. I thought to keep things simpler it might be better to have all changes in `pkg/util/filepath.go`, but happy to change some of the walker code and have this here if it's easier to reason about.

Reply:
I _think_ it's better to have this logic here because:

 - `walk` is already quite complex already and in theory is generic (based on it's path). Since the `dist` customization is plugin-specific, sounds to me this is a better place.
 - The tests are here, in this package. If not it may look like the logic in `walk` is not covered.

I don't know if we can get away with no changes in `filepath.go` if we have this logic though. Would the code be split in two places in that case?

Reply:
>Since the dist customization is plugin-specific, sounds to me this is a better place.

>I don't know if we can get away with no changes in filepath.go if we have this logic though. Would the code be split in two places in that case?

The thing is though there's logic already in there for `dist`, and  yeah we would have to have it in two separate places because we're not able to specify the "please just walk the dist only and nothing else". It's just a bit confusing mixing the logic, but not sure it's much worse than what we currently have 🤷

Reply:
`util.Walk` seems like it's only used by plugins so why not move walk function into this package?

Reply:
Yep that's the plan! Saving that for a follow up 👍

Reply:
Okay. Then I think these comments can be removed because they're confuse more than help right now I think :)